src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -76,6 +77,31 @@ class Parameter
         // if the parameter hasn't been registered yet, return default value
         return this->value_;
     }
+
+    /**
+     * Returns true if the value has recently changed, and false otherwise
```

Comment:
What does "recently" mean here?

Reply:
It's a _bit_ of a hack. It means the value was changed during the current "tick" of the AI. This "value_recently_changed" flag gets cleared when the value is read, so it assumes the variable is only being read once / from one place. I've added a more descriptive comment.

Reply:
I'm really not a big fan of this. I think I'd rather see us maintain state in `STP` then add a hack here.

Reply:
Depending on how we decide to use this function it may be misleading because IIRC every parameter will update with every parameter change. If you change one parameter, it will call send 1 message with all the current values to the callback, which then sets all the parameters. 

So it may not be useful if the goal is to do something if the value changes, because this flag could flip without the value actually changing (at least for the callback method). 

But if this is the behavior that we would like maybe we need to change the name of the function from `valueUpdated()` to something else.

Reply:
Fair enough. I've reverted the parameter changes and now have STP store the values to detect changes.